Lib/test/seq_tests.py

```diff
@@ -250,6 +250,43 @@ def __eq__(self, other):
         checklast = self.type2test([StopCompares(), 1])
         self.assertRaises(DoNotTestEq, checklast.__contains__, 1)
 
+    def test_compare_nan(self):
+        nan = float('nan')
+        a = self.type2test([nan])
+        b = self.type2test([nan])
+        self.assertTrue(a == a)
+        self.assertFalse(a != a)
+        self.assertTrue(a == b)
+        self.assertFalse(a != b)
+        self.assertFalse(a < a)
+        self.assertFalse(a < b)
+        self.assertFalse(a > b)
```

Comment:
This code seems to be copied/pasted in 3 functions. Can you add a helper function, such as "def check_compare_id(self, a, b)` to factorize the code?